File: OpenValidationFramework-Python/test/rule_test.py
import HUMLFramework as huml 

class TestRulesAppendRule:

    def test_appending_a_rule(self):
        validator = huml.HUMLFramework()
        validator.append_rule(
            name="",
            fields=["name"],
            error="name must be something",
            condition_function=lambda input, cntx: huml.NOT_EQUALS(input, "something"),
            disabled=False
        )
        assert len(validator.rules) >= 1

    def test_appending_10_rules(self):
        validator = huml.HUMLFramework()
        for _ in range(10):
            validator.append_rule(
                "",
                ["name"],
                "name must be something",
                lambda input, cntx: huml.NOT_EQUALS(input, "something"),
                False
            )
        assert len(validator.rules) >= 10

class TestRulesValidate:

    def test_none_not_equals_something(self):
        validator = huml.HUMLFramework()
        validator.append_rule(
            "",
            ["name"],
            "name must be something",
            lambda input, cntx: huml.NOT_EQUALS(input, "something"),
            False
        )
        assert validator.validate(None)["has_errors"]

File: OpenValidationFramework-Javascript/test/rule-test.js
describe('Rules - append rule', () => { 
        it('append rule -> rules length should be greater or equal to 1', () => { 
            var hml = new HUMLFramework();
            hml.appendRule("",
                ["name"],
                "name must be something",
                function(input) { 
                    return huml.NOT_EQUALS(input, "something");
                },
                false
            );

            expect(huml.GREATER_OR_EQUALS(hml.rules.length, 1)) 
                .toEqual(true); 
        });

        it('append duplicate rule -> rules length should be greater or equal to 10', () => { 
            var hml = new HUMLFramework();
            for(var i = 0;i<10;i++){
                hml.appendRule("",
                    ["name"],
                    "name must be something",
                    function(input) { 
                        return huml.NOT_EQUALS(input, "something");
                    },
                    false
                );
            }

            expect(huml.GREATER_OR_EQUALS(hml.rules.length, 10)) 
                .toEqual(true); 
        });
});

describe('Rules - validate rule', () => { 
    it('validate rule -> nothing NOT_EQUALS something should be true', () => { 
        var hml = new HUMLFramework();
        hml.appendRule("",
            ["name"],
            "name must be something",
            function(input) { 
                return huml.NOT_EQUALS(input, "something");
            },
            false
        );

        expect(huml.validate(null).hasErrors).toEqual(false);
    });     

    it('validate rule -> EMPTY should be true', () => { 
        var hml = new HUMLFramework();
        hml.appendRule("",
            ["name"],
            "name must be something",
            function(input) { 
                return huml.EMPTY(input.name);
            },
            false
        );

        var mod = {'name':'hans'};
        var res = huml.validate(mod);
        expect(res.hasErrors).toEqual(false);
    }); 
});